src/SDK/MonitoringApplianceEmulator/Models/ObservableTask{TResult}.cs

```diff
@@ -0,0 +1,150 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ObservableTask{TResult}.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Azure.Monitoring.SmartDetectors.MonitoringApplianceEmulator.Models
+{
+    using System;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Occurs when the observable task was completed.
+    /// </summary>
+    /// <typeparam name="T">The type of the object that will be passed as an input parameter.</typeparam>
+    /// <param name="obj">The input parameter.</param>
+    public delegate void OnTaskCompletedEventHandler<in T>(T obj);
+
+    /// <summary>
+    /// An observable task, inherits from <see cref="ObservableObject"/>.
+    /// </summary>
+    /// <typeparam name="TResult">The type of the result being returned from the task. This type should have default constructor,
```

Comment:
>This type should have default constructor [](start = 87, length = 41)

I don't think that's true - you just need a default value, not necessarily a default constructor

Reply:
good catch

---
In reply to: [231003629](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/13#discussion_r231003629) [](ancestors = 231003629)